src/arch/x86/tlb.cc

```diff
@@ -94,8 +94,11 @@ TLB::evictLRU()
 }
 
 TlbEntry *
-TLB::insert(Addr vpn, const TlbEntry &entry)
+TLB::insert(Addr vpn, const TlbEntry &entry, int asn)
 {
+
+    vpn = vpn | (asn & 1);
```

Comment:
Are we guaranteed that `vpn & 1 == 0`? Should we left shift `vpn`?

Either way, this should be a separate (inline) function because there's going to be many places that we need to construct this tag.

Reply:
I have changed this implementation a bit. But, my main idea is that since, vpn here is a page aligned virtual address (0 ' bits are not trimmed), performing an OR operation of this address with the process id (assuming we will not have a process id greater than 2^(no. of bits used for page offset) should lead to a unique number for each (virtual address + process)  pair. The other change needed for this to work is change the number of bits used in trie.insert to be MaxBits.